File: iosApp/iosApp/service/DashboardService.swift
//
// Created by Tim Ortel on 27.09.22.
//
//

import Foundation

/**
 * Service where you can make requests about the Artemis dashboard.
 */
protocol DashboardService {

    /**
     * Load the dashboard from the specified server using the specified authentication data.
     */
    func loadDashboard(authorizationToken: String, serverUrl: String) async -> NetworkResponse<Dashboard>
}

File: appCommon/src/commonMain/kotlin/de/tum/informatics/www1/artemis/native_app/service/DashboardService.kt
package de.tum.informatics.www1.artemis.native_app.service

import de.tum.informatics.www1.artemis.native_app.content.Dashboard
import de.tum.informatics.www1.artemis.native_app.util.DataState
import de.tum.informatics.www1.artemis.native_app.util.NetworkResponse
import kotlinx.coroutines.flow.Flow

/**
 * Service where you can make requests about the Artemis dashboard.
 */
interface DashboardService {

    /**
     * Load the dashboard from the specified server using the specified authentication data.
     */
    suspend fun loadDashboard(authenticationData: AccountService.AuthenticationData.LoggedIn, serverUrl: String): NetworkResponse<Dashboard>
}